tools/reactive-controllers/test/pending-state.test.ts

```diff
@@ -0,0 +1,153 @@
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+/*
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { expect, fixture, html } from '@open-wc/testing';
+import {
+    HostWithPendingState,
+    PendingStateController,
+} from '@spectrum-web-components/reactive-controllers/src/PendingState.js';
+import sinon from 'sinon';
+import '@spectrum-web-components/progress-circle/sp-progress-circle.js';
+import '@spectrum-web-components/picker/sp-picker.js';
+
+describe('PendingStateController', () => {
+    let host: HostWithPendingState;
+    let controller: PendingStateController<HostWithPendingState>;
+
+    beforeEach(async () => {
+        host = await fixture<HostWithPendingState>(html`
+            <sp-picker aria-label="clickable" pending></sp-picker>
+        `);
+        controller = host.pendingStateController;
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    describe('renderPendingState', () => {
+        it('should change aria-label of host when pending and when not pending', async () => {
+            await host.updateComplete;
+            let ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('Pending');
+
+            host.removeAttribute('pending');
+            await host.updateComplete;
+            ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('clickable');
+
+            host.setAttribute('pending', 'true');
+            await host.updateComplete;
+            ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('Pending');
+        });
+
+        it('should render the pending state UI', async () => {
+            const pendingLabel = 'Custom Pending Label';
+            host.pendingLabel = pendingLabel;
+            const templateResult = controller.renderPendingState();
+
+            const renderedElement = await fixture(html`
+                ${templateResult}
+            `);
+            const expectedElement = await fixture(html`
+                <sp-progress-circle
+                    id="loader"
+                    size="s"
+                    indeterminate
+                    aria-valuetext=${pendingLabel}
+                    class="progress-circle"
+                ></sp-progress-circle>
+            `);
+
+            expect(renderedElement.outerHTML === expectedElement.outerHTML).to
+                .be.true;
+        });
+
+        it('should render the default pending state UI if no label is provided', async () => {
+            host.pendingLabel = undefined;
+            const templateResult = controller.renderPendingState();
```

Comment:
Here we are testing the `controller.renderPendingState()` in isolation, which is great. Should we have a test on the `button.test.ts` that checks for the expected markup there when the component is `pending`?